### DISCUSSION TITLE
A centralized place to chat!

### DISCUSSION THREAD

**Discussion by spiderman82** [truckersmp-cli/truckersmp-cli#279] [Ideas]
Greetings to Kakurasan! I recently returned to ETS 2 multiplayer. And I couldn't help but notice that the topic you created on the forum about multiplayer launch on Linux was closed. I would dare to suggest other ways of our (those who use Linux) communication, for example, Discord! I think this will be more comfortable than Github ...

**Comment by Lucki** [comment]
Greetings to Kakurasan! I've activated the discussions. Hopefully this is already fitting but we can evaluate Discord and Matrix later on.

**Reply by kakurasan** [reply]
Greetings to everyone!

> we can evaluate Discord and Matrix later on.

Unfortunately I don't know much about Discord/Matrix. Is anyone interested in setting up?

**Reply by Lucki** [reply]
> Is anyone interested in setting up?

This is not something we should give out of hand.

Another possibility is a Steam group with the associated chat rooms.

**Comment by spiderman82** [comment]
https://discord.gg/7KqKrUjE for a trial you can try my channel

**Comment by Lucki** [comment]
While I personally prefer Matrix it doesn't seem reasonable for a general gaming related group.
This is where Discord fits but the more I thought about it I actually don't get why we should direct the group to a completely unrelated service.

And this is where Steam comes in. It is needed anyway to play the game and like Discord it
* can show ingame status (even rich but unsupported by truckersmp, ets, ats)
* has mobile and web clients

and even more like
* forum
* events

Compared to Discord it's lacking some advanced features like an API or chat history search.

Anyway, I've set up a public Steam group with the associated public chat - feel free to join and chat together!

* Group: https://steamcommunity.com/groups/truckersmp-cli
* Chat: https://s.team/chat/Afb9Z3Ox

**Reply by spiderman82** [reply]
Discord at least has a good voice chat, you can share the screen!

**Comment by spiderman82** [comment]
about chat...
![Screenshot_20220116_092829](https://user-images.githubusercontent.com/35892244/149649918-7e8543c6-9faf-4b30-bfea-0d8e0c61a0f7.png)

**Comment by KeyboardZombie** [comment]
Is there a clearer tutorial for this? I'm trying to setup truckers mp cli on my steam deck, but its not very clear to linux newbies.

**Reply by Lucki** [reply]
I don't have a steam deck so I can't give you the exact steps needed. I've heard the root system is immutable so you have to use the per user installation approach either through [pythons PPI](https://github.com/truckersmp-cli/truckersmp-cli#python-package-index) (if that's available on the steam deck) or through a [manual download](https://github.com/truckersmp-cli/truckersmp-cli#manual-download).

You'll need a command line at least for the initial setup and updates so you probably want to be in desktop mode for this.
1. Install truckersmp-cli, choose one option:
    * `python3 -m pip install --user truckersmp-cli[optional]`
    * Download and unpack the [latest release](https://github.com/truckersmp-cli/truckersmp-cli/releases/latest), found in the _Assets_ section
1. Navigate to the main binary file:
    * `cd $HOME/.local/bin` for python user installation
    * `cd $HOME/my/unpacked/directory` for manual download
1. Run `pwd` and remember the path for the steam shortcut creation later.
1. Run `./truckersmp-cli update ets2mp` to install ETS2 and TruckersMP
    You might have to append your steam account name based on your chosen installation method: `--account NAME`
    This will take a while based in your connection speed.
    Repeat this step if you want to update ETS2 in the future.
    This will close steam so don't be surprised if some controls behave differently.
1. You can now start the game with `./truckersmp-cli start ets2mp --rendering-backend dx11`
1. In the future you can update this program, choose based on the installation above:
    * `python3 -m pip install --user truckersmp-cli --upgrade`
    * `./truckersmp-cli update ets2mp --self-update`

If the above worked you can proceed.
To start the game from the touch UI mode you have to add a shortcut to steam: (This will only work if steam itself isn't a Flatpack)
1. Add an external program shortcut to steam - choose a random one that's available.
4. Go into it's properties to change some values:
    * Change the _target_ to the truckersmp-cli binary you've located before. You'll have to specify the full path here, remembered from before.
      `/home/user/path/to/truckersmp-cli`
    * Change the launch options to the desired values
      `start ets2mp --rendering-backend dx11`

I hope this is detailed enough and will work on the deck.
